thewalrus/tests/test_entanglement.py

```diff
@@ -0,0 +1,158 @@
+# Copyright 2022 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Entanglement tests"""
+# pylint: disable=no-self-use, assignment-from-no-return
+import pytest
+
+import numpy as np
+
+from thewalrus.quantum import entanglement_entropy, log_negativity
+from thewalrus.random import random_interferometer
+from thewalrus.symplectic import (
+    interferometer,
+    passive_transformation,
+    two_mode_squeezing,
+    xxpp_to_xpxp,
+)
+
+
+def random_cov(num_modes=200, pure=False, nonclassical=True):
+    r"""Creates a random covariance matrix for testing.
+
+    Args:
+        num_modes (int): number of modes
+        pure (bool): Gaussian state is pure
+        nonclassical (bool): Gaussian state is nonclassical
+
+    Returns:
+        (array): a covariance matrix
+    """
+    M = num_modes
+    O = interferometer(random_interferometer(M))
+
+    if pure:
+        eta = 1
+    elif not pure:
+        eta = 0.123
+
+    r = 1.234
+    if nonclassical:
+        # squeezed inputs
+        cov_in = np.diag(np.concatenate([np.exp(2 * r) * np.ones(M), np.exp(-2 * r) * np.ones(M)]))
+    elif not nonclassical and pure:
+        # vacuum inputs
+        cov_in = np.eye(2 * M)
+    elif not nonclassical and not pure:
+        # squashed inputs
+        cov_in = np.diag(np.concatenate([np.exp(2 * r) * np.ones(M), np.ones(M)]))
+
+    cov_out = O @ cov_in @ O.T
+    _, cov = passive_transformation(
+        np.zeros([len(cov_out)]), cov_out, np.sqrt(eta) * np.identity(len(cov_out) // 2)
+    )
```

Comment:
```suggestion
    T = np.sqrt(eta) * np.identity(len(cov_out) // 2
    _, cov = passive_transformation(np.zeros([len(cov_out)]), cov_out, T))
```

Reply:
Seems like the `mu` perhaps should be optional in `passive_transformation`, so as to not have to pass a dummy-mu here.

Reply:
agreed, but that would require a significant change in many places, maybe this could be added as an issue.

Reply:
Done: https://github.com/XanaduAI/thewalrus/issues/327